### PR TITLE
Redux Store

### PR DESCRIPTION
**Pull request by flxffy**
Sets up persistent redux store :partying_face: 

Actions and reducers defined in `src/redux`

- Using [reduxsauce](https://github.com/jkeam/reduxsauce#readme) to create actions and map actions to reducers

Sample usage can be found in `components/PlanDropdown`
To reinitialise the store, clear the persisted store from `localStorage`

Store:
```
{
  currentPlan: "string",
  plans: {
    [plan]: {
      sems: {
        [sem]: {
          name: "string",
          mods: [{
            code: "string",
            name: "string",
            mcs: 4
          }]
        }
      },
      specialisations: ["string"],
      majors: ["string"],
      minors: ["string"]
    }
  }
}
```

### CODE REVIEW

**Review by kenneth-fung — APPROVED** [review]
I kinda think I kinda understand what you did? Looks good though. Thanks a lot for redux integration.

File: src/components/PlanDropdown.jsx [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
 
+import { Actions } from "../redux/actions";
 import EditableList from "./EditableList";
 
 import "../scss/Dropdown.scss";
 
-const PlanDropdown = ({ plans = ["Second Major", "without honours"], selected = "Default Plan" }) => {
+const PlanDropdown = ({ plans, selected, handleDelete, handleChangePlan }) => {
   const history = useHistory();
+  const items = Object.keys(plans).filter((plan) => plan !== selected);
```
Comment (hellodommy) [inline src/components/PlanDropdown.jsx:15]: great abstraction

File: src/components/PlanDropdown.jsx [hunk 2]
```diff
@@ -38,4 +41,14 @@ const PlanDropdown = ({ plans = ["Second Major", "without honours"], selected =
   );
 };
 
-export default PlanDropdown;
+const mapStateToProps = (state) => ({
```
Comment (hellodommy) [inline src/components/PlanDropdown.jsx:44]: Does this mean we'll know what plan the user is using?

File: src/data/computing-modules-detailed.json [hunk 3]
```diff
@@ -5837,5 +5837,20 @@
     "hasPrerequisite": true,
     "url": "https://nusmods.com/modules/XFC4102/integrated-honours-dissertation",
     "semesterData": []
+  },
+  {
+    "moduleCode": "ES2660",
```
Comment (hellodommy) [inline src/data/computing-modules-detailed.json:5842]: My favourite mod

File: src/redux/reducers.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+export const changePlan = (state = INITIAL_STATE, { name }) => {
+  return { ...state, currentPlan: name };
+};
+
+export const addPlan = (state = INITIAL_STATE, { plan }) => {
+  const plans = { ...state.plans, ...plan };
+  return { ...state, plans };
+};
+
+export const renamePlan = (state = INITIAL_STATE, { prevName, newName }) => {
```
Comment (hellodommy) [inline src/redux/reducers.js:20]: Is there a popup in place to rename?

**Review by hellodommy — APPROVED** [review]

File: src/components/PlanDropdown.jsx (hunk 2 above)
Comment (kenneth-fung) [reply src/components/PlanDropdown.jsx:44]: Yea it'll be passed into the `PlanDropdown` component state as the `selected` plan.

File: src/redux/reducers.js (hunk 4 above)
Comment (kenneth-fung) [reply src/redux/reducers.js:20]: Not yet I think, but the action is there so we could use it in the future.

**Pull request merged**